### ISSUE TITLE
Syntax error on nested ternary expressions

### ISSUE THREAD

**Issue by XmiliaH** [PlutoLang/Pluto#782]
Nesting a ternary espression might not work. Is it intended that nested ones need to be in parentesis?

```lua
local a = b ? c ? d : e : f
```

current fix

```lua
local a = b ? (c ? d : e) : f
```

**Comment by Sainan** [comment]
Yes, when reading the false-expression, it gets parsed like so:
```lua
local a = b ? c ? d : e:f
```
so it's now expecting function arguments for the call to `e:f`. Pluto also warns you about this:
![image](https://github.com/PlutoLang/Pluto/assets/63328889/a3a7d381-7c0d-48cb-aff3-ef5756b7264f)

**Comment by XmiliaH** [comment]
Here is another case:
```lua
local a = b ? c in d : e
```
why is `:` here parsed as method invocation and without the `in` as the end of the ternary? Seems not very consistent to me.

**Comment by Sainan** [comment]
That one I'd say is a bug; fixed by #784.
